flake.nix

```
@@ -0,0 +1,87 @@
+{ inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/release-22.11";
+
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+
+    flake-utils.url = "github:numtide/flake-utils/v1.0.0";
+  };
+
+  outputs = { nixpkgs, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        ghcVersion = "ghc90";
+
+        config = { };
+
+        overlay = self: super: {
+          haskell = super.haskell // {
+            packages = super.haskell.packages // {
+              "${ghcVersion}" = super.haskell.packages."${ghcVersion}".override (old: {
+                overrides =
+                  self.lib.composeExtensions
+                    (old.overrides or (_: _: { }))
+                    (hself: hsuper: {
+                      # The reason we stick these underneath a `local` attribute
+                      # is to avoid infinite recursion (because `callCabal2nix`
+                      # depends on all of these packages.
+                      local = rec {
+                        cabal2nix =
+                          hself.callCabal2nix "cabal2nix" ./cabal2nix {
+                            inherit distribution-nixpkgs hackage-db;
```

This would also depend on `language-nix`.

```suggestion
                            inherit distribution-nixpkgs hackage-db language-nix;
```